### PR TITLE
feat: add trimCharsStart

### PR DESCRIPTION
**Pull request by kitasuna**
#1135 

Add trimCharsStart function / types / tests. Happy to receive any feedback 🙇

### CODE REVIEW

File: src/trimCharsStart.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @sig String -> String
+ * @param {string} value String that contains all characters to be removed from the begining of the string provided in the second argument
+ * @param {string} value String value which will have the characters specified by the first argument removed from its beginning
+ * @return {string} A new string representing the second argument with the specified characters removed from its beginning (left end).
+ * @example
+ *
+ * RA.trimCharsStart('_-', '-_-abc-_-'); //=> 'abc-_-'
+ */
+
+const trimCharsStart = curry((charsToTrim, str) =>
```
Comment (char0n) [inline src/trimCharsStart.js]: charsToTrim -> chars
str -> value

File: src/trimCharsStart.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  join(
+    '',
+    dropWhile(x => contains(x, charsToTrim), split('', str))
```
Comment (char0n) [inline src/trimCharsStart.js]: Let's use functional composition and [RA.contained](https://char0n.github.io/ramda-adjunct/2.23.0/RA.html#.contained) function
```js
pipe(
  split(''),
  dropWhile(contained(chars)),
  join('')
)(value)
```

File: test/trimCharsStart.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { assert } from 'chai';
+
+import * as RA from '../src';
+
+describe('trimCharsStart', function() {
+  it('should remove all occurences of a trim character', function() {
```
Comment (char0n) [inline test/trimCharsStart.js]: Pls don't use currying in tests. We'll have a separate test that is testing currying itself.
```js
assert.strictEqual(RA.trimCharsStart('-', '---abc'), 'abc');

File: test/trimCharsStart.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  it('should remove all occurences of all trim characters', function() {
+    assert.strictEqual(RA.trimCharsStart('_-')('-_-abc'), 'abc');
+  });
+
+  it('should not remove the trim characters from the end of the string', function() {
+    assert.strictEqual(RA.trimCharsStart('_-')('-_-abc-_-'), 'abc-_-');
+  });
+
+  it('should return the target string as-is if none of the trim characters are present', function() {
```
Comment (char0n) [inline test/trimCharsStart.js]: If test descriptions gets long it's very hard to read them. We have a convention to fix that
```js
context('given none of the trim characters are present', function() {
  specify('should return the target string as-is ', function() {
    assert.strictEqual(RA.trimCharsStart('_-', 'abc'), 'abc');
  });        
});

Comment (char0n) [reply test/trimCharsStart.js]: Usualy the long sentences cotains logical branching (if/when). Please process the rest of the tests as I described above.

File: test/trimCharsStart.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    assert.strictEqual(RA.trimCharsStart('_-')('-_-_-_'), '');
+  });
+
+  it('should return the target string as-is if given an empty string for the trim characters', function() {
+    assert.strictEqual(RA.trimCharsStart('')('abc'), 'abc');
+  });
+
+  it('should return empty string if the target string is also empty string', function() {
+    assert.strictEqual(RA.trimCharsStart('-_')(''), '');
+  });
```
Comment (char0n) [inline test/trimCharsStart.js]: We're missing test for currying
```js
  it('should curry', function() {
    assert.strictEqual(RA.trimCharsStart('-_', ''), '');
    assert.strictEqual(RA.trimCharsStart('-_')(''), '');
  });

File: types/index.d.ts [hunk 6]
```diff
@@ -1294,6 +1294,12 @@ declare namespace RamdaAdjunct {
         trimEnd(value: string): string;
         trimRight(value: string): string;
 
+        /**
+         * Removes specified characters from the beginning of a string
+         */
+        trimCharsStart(charsToTrim: string, targetString: string): string;
```
Comment (char0n) [inline types/index.d.ts]:
```suggestion
        trimCharsStart(chars: string, value: string): string;
```

File: types/index.d.ts [hunk 7]
```diff
@@ -1294,6 +1294,12 @@ declare namespace RamdaAdjunct {
...
+        trimCharsStart(charsToTrim: string): (targetString: string) => string;
```
Comment (char0n) [inline types/index.d.ts]:
```suggestion
        trimCharsStart(chars: string): (value: string) => string;
```

File: types/index.d.ts [hunk 8]
```diff
@@ -1294,6 +1294,12 @@ declare namespace RamdaAdjunct {
         trimEnd(value: string): string;
         trimRight(value: string): string;
 
+        /**
+         * Removes specified characters from the beginning of a string
```
Comment (char0n) [inline types/index.d.ts]:
```suggestion
         * Removes leading specified characters from string.
```

**Comment by kitasuna** [comment]
Thanks very much for the quick review and feedback! I've attempted to address your comments, and committed the suggestions you left as well.

**Comment by kitasuna** [comment]
Not sure what to make of this message from CI:

```
not ok 3133 Chrome - [undefined ms] - error
    ---
        message: >
            Error: Browser timeout exceeded: 10s
            Stderr: 
             Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
            [8840:8877:0107/143103.038344:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
            [8840:8971:0107/143103.120100:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
            ATTENTION: default value of option force_s3tc_enable overridden by environment.
            
            (google-chrome:8840): LIBDBUSMENU-GLIB-WARNING **: Unable to get session bus: Unknown or unsupported transport 'disabled' for address 'disabled:'
            [8891:8891:0107/143103.170580:ERROR:sandbox_linux.cc(368)] InitializeSandbox() called with multiple threads in process gpu-process.
            [8985:9014:0107/143103.184619:ERROR:command_buffer_proxy_impl.cc(126)] ContextResult::kTransientFailure: Failed to send GpuChannelMsg_CreateCommandBuffer.
            
            
        browser log: |
            [object Object]
            [object Object]
    ...
```

**Comment by char0n** [comment]
@kitasuna I've restarted the tests. This sometimes happens. Some internals of CircleCI fails and it reports as false positive. Should be ok after tests finish running.

File: test/trimCharsStart.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+  it('should remove all occurences of all trim characters', function() {
+    assert.strictEqual(RA.trimCharsStart('_-', '-_-abc'), 'abc');
+  });
+
+  it('should not remove the trim characters from the end of the string', function() {
+    assert.strictEqual(RA.trimCharsStart('_-', '-_-abc-_-'), 'abc-_-');
+  });
+
+  it('should curry', function() {
```
Comment (char0n) [inline test/trimCharsStart.js]: can you pls move this test to be the last  ? We have this convention of moving currying tests to be last. Thank you!

**Comment by kitasuna** [comment]
@char0n I moved the curry test to the end and rebased. Thanks for the guidance in getting this up to snuff! It's been fun to work on.

**Comment by char0n** [comment]
> I moved the curry test to the end and rebased

@kitasuna, thank you! Next time no need for you to rebase, GitHub has a feature `Squash and merge` that allows me to squash and merge your PR it into single commit. In past the contributors were required to rebase to clean up the PR, but it's no longer required as I can do it before the merge itself directly on Github webpage.

> Thanks for the guidance in getting this up to snuff! It's been fun to work on.

My pleasure! In future if you want to continue with work here take a look at all the [issues you could implement](https://github.com/char0n/ramda-adjunct/issues?utf8=%E2%9C%93&q=is%3Aopen+label%3A%22help+wanted%22+label%3Afeature). Maybe some of them will interest you.

**Comment by char0n** [comment]
I'll create a release for this function during today/tomorrow.

**Pull request merged**